host/xtest/xtest_uuid_helpers.c

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2019, Linaro Limited
+ */
+
+#include <ctype.h>
+#include <stdint.h>
+#include <string.h>
+#include <endian.h>
+#include <openssl/evp.h>
+#include <tee_api_types.h>
+#include <tee_client_api.h>
+#include "xtest_uuid_helpers.h"
+
+static int hex(char c)
+{
+	char lc = tolower(c);
+
+	if (isdigit(lc))
+		return lc - '0';
+	if (isxdigit(lc))
+		return lc - 'a' + 10;
+	return -1;
+}
+
+static uint32_t parse_hex(const char *s, size_t nchars, uint32_t *res)
+{
+	uint32_t v = 0;
+	size_t n = 0;
+	int c = 0;
+
+	for (n = 0; n < nchars; n++) {
+		c = hex(s[n]);
+		if (c == -1) {
+			*res = TEE_ERROR_BAD_FORMAT;
+			goto out;
+		}
+		v = (v << 4) + c;
+	}
+	*res = TEE_SUCCESS;
+out:
+	return v;
+}
+
+TEEC_Result xtest_uuid_from_str(TEEC_UUID *uuid, const char *s)
+{
+	TEEC_Result res = TEEC_SUCCESS;
+	TEEC_UUID u = { };
+	const char *p = s;
+	size_t i = 0;
+
+	if (!p || strnlen(p, 37) != 36)
+		return TEEC_ERROR_BAD_FORMAT;
+	if (p[8] != '-' || p[13] != '-' || p[18] != '-' || p[23] != '-')
+		return TEEC_ERROR_BAD_FORMAT;
+
+	u.timeLow = parse_hex(p, 8, &res);
```

Comment:
suggest:
```
static TEEC_Result advance_dash(const char **s)
{
	if (**s != '-')
		return TEEC_ERROR_BAD_FORMAT;

	(**s)++;

	return TEEC_SUCCESS;
}

static TEEC_Result parse_hex32_and_adance(const char **s, size_t nchars, uint32_t *dst)
{
	uint32_t v = 0;
	size_t n = 0;
	int c = 0;

	for (n = 0; n < nchars; n++) {
		c = hex((*s)[n]);
		if (c == -1)
			return TEE_ERROR_BAD_FORMAT;

		v = (v << 4) + c;
	}

	*dest = v;
	*s += nchars;

	return TEE_SUCCESS;
}

TEEC_Result xtest_uuid_from_str(TEEC_UUID *uuid, const char *s)
{	TEEC_UUID u = { };
	const char *p = s;
	size_t i = 0;

	if (get_hex32_and_advance(&p, 8, &u.timeLow)
	    advance_dash(&p) ||
	    get_hex32_and_advance(&p, 8, &u.timeMid)
	    advance_dash(&p) ||
	    get_hex32_and_advance(&p, 4, &u.timeHiAndVersion)
	    advance_dash(&p))
		return TEEC_ERROR_BAD_FORMAT;

	for (i = 0; i < 8; i++) {
		if (get_hex32_and_advance(&p, 2, u.clockSeqAndNode + i))
			return TEEC_ERROR_BAD_FORMAT;

		if (i < 7 && advance_dash(&p))
			return TEEC_ERROR_BAD_FORMAT;
	}

	*uuid = u;

	return TEE_SUCCESS;
}

Reply:
Is it easier to read? I don't think so. And, the commit simply imports existing code, there is no reason to change it IMO.

Reply:
Left as is.